frontend/src/lib/user.ts

```diff
@@ -115,7 +115,25 @@ function clear(): void {
 
 export async function hash(password: string): Promise<string> {
   const msgUint8 = new TextEncoder().encode(password) // encode as (utf-8) Uint8Array
-  const hashBuffer = await crypto.subtle.digest('SHA-1', msgUint8) // hash the message
+  let hashBuffer: ArrayBuffer;
+  // const c = crypto ? crypto : await import('node:crypto');
```

Comment:
```suggestion
```

Reply:
I'm actually thinking I'll put this line in and remove the more elaborate try...catch block, because if `import('node:crypto')` fails then it will throw an exception, which is just what you were suggesting in the other comment.

BTW, the reason for this is because we were running into `https://github.com/nodejs/node/issues/48444` during our docker build process, so I had to temporarily downgrade the Node version to 18 instead of 20 to work around that bug — and Node 18 doesn't have the same crypto modules. This workaround will be able to go away soon, though: that issue is affecting a lot of people so they're fast-tracking a new release of the affected library. At which point we'll be able to go back to building with modern Node versions and this hack can be removed.